### PR TITLE
feat: create client package and update code

### PR DESCRIPTION
**Pull request by sudo-sturbia**
Hi! I did a couple of small changes to tidy the package a bit, and added a client package, as the main package can't be imported. Feedback is welcomed!

I plan on applying for GSoC with casbin, and want to contribute to see how things work. I thought I'd be useful here as the repo doesn't have a lot of contributions. I also plan to work on extending this package to allow more of the casbin API. Hopefully, this can be considered for my application!

### CODE REVIEW

**Comment by hsluoyz** [comment]
@sudo-sturbia the git history shows the files are removed and then added, the history is missing. Plz fix it.

**Comment by sudo-sturbia** [comment]
@hsluoyz This is happening because the files are moved into a new directory in 1736fb2. It may be easier to review changes one commit at a time.

File: client/client.go [hunk 1]
```diff
@@ -11,7 +13,7 @@ type Client struct {
 	remoteClient pb.CasbinClient
 }
 
-func NewClient(ctx context.Context, address string, opts ...grpc.DialOption) (*Client, error) {
+func New(ctx context.Context, address string, opts ...grpc.DialOption) (*Client, error) {
```
Comment (hsluoyz) [inline client/client.go]: Still naming it as `NewClient` would be OK.

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
> @hsluoyz This is happening because the files are moved into a new directory in [1736fb2](https://github.com/casbin/casbin-go-client/commit/1736fb26b7ca1444a537c9012ab923365deb85f4). It may be easier to review changes one commit at a time.

OK.

File: client/client.go (hunk 1 above)
Comment (sudo-sturbia) [reply client/client.go]: Done

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**